### ISSUE TITLE
SSH disconnection: broken pipe

### ISSUE THREAD

**Issue by lherman-cs** [ayufan-rock64/linux-build#51]
I always get a random broken pipe from my rock64. Below is the message:

getaddrinfo: Servname not supported for ai_socktype
packet_write_wait: Connection to UNKNOWN port 0: Broken pipe

sshd_config (Server side):
ClientAliveInterval 20
ClientAliveCountMax 5
TCPKeepAlive no

ssh_config (Client side):
ServerAliveInterval 120

**Comment by StuartIanNaylor** [comment]
Been using quite a few of the different images and ssh has been solid for me with the 118 release versions.
Prob need to state which image.

**Comment by lherman-cs** [comment]
Oh, that's right. I forgot to mention that. Following is the result after I run "uname -a":

Linux rock64 4.4.77-rockchip-ayufan-118 #1 SMP Thu Sep 14 21:59:24 UTC 2017 aarch64 aarch64 aarch64 GNU/Linux

**Comment by mitag** [comment]
I'm not sure if it helps, but I had the same issue with an image from the Rock64 main page.
(Stock Build Debian Jessie LXDE Linux Kernel ver 4.4 [microSD Boot] [20170809] Release version)
Later I downloaded and install your xenial-mate-rock64-0.5.10-118-arm64_sd2emmc.img without this issue.

**Comment by lherman-cs** [comment]
Did you customize your ssh config or sshd config?

**Comment by mitag** [comment]
No, I did not customize the configuration of ssh or sshd. I tried ssh under Windows with ConEmu and Cygwin_64 ssh and with Debian Stretch ssh.

**Comment by naip1984** [comment]
I attempted the work arounds above... 

I ran into the same problem attempting to use: 

Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.3 LTS
Release:	16.04
Codename:	xenial

with distros: 

1.   4.4.103-rockchip-ayufan-164 #1 SMP Sat Jan 6 11:15:18 UTC 2018 aarch64 aarch64 aarch64 GNU/Linux

2. 4.4.77-rockchip-ayufan-101 #1 SMP Fri Sep 8 14:48:39 UTC 2017 aarch64 aarch64 aarch64 GNU/Linux

3.  4.4.77-rockchip-ayufan-118 #1 SMP Thu Sep 14 21:59:24 UTC 2017 aarch64 aarch64 aarch64 GNU/Linux

The physical devices were all different for each attempt (different rock 64 boards).  

I noticed that this only seemed to happen when I had an attached USB external drive.   

I started seeing errors like these in dmesg:   

dev_pm_opp_get_voltage: Invalid parameters
rcu_sched kthread starved for 59999 jiffies! g11016 c11015 f0x2 s4 ->state=0x40
INFO: rcu_sched detected stalls on CPUs/tasks:
mmcblk1: error -110 sending status command, aborting 

sometimes the problem would result in a KP,  other times the connection would be lost and the transfer would fail.   

I googled around a bit, about found other people on other distros complaining about UBS 3.0 problems.     So I switched to use a UBS 2 port.     The problem went away.   

So, I think there is a USB 3.0 issue that could also be causing this problem.     

There's a few posts on Japanese Rock64 forums that (translated by google translate) lead me to believe some is working on the problem, but I have found any further updates.   Does anyone have anymore info about USB 3.0 issues using ROCK64 boards?
